crates/bevy_ecs/src/world/mod.rs

```diff
@@ -46,6 +50,28 @@ pub struct World {
     /// Access cache used by [WorldCell].
     pub(crate) archetype_component_access: ArchetypeComponentAccess,
     main_thread_validator: MainThreadValidator,
+    pub(crate) global_system_counter: AtomicU32,
+    pub(crate) exclusive_system_counter: u32,
+}
+
+impl Default for World {
+    fn default() -> Self {
+        Self {
+            id: Default::default(),
+            entities: Default::default(),
+            components: Default::default(),
+            archetypes: Default::default(),
+            storages: Default::default(),
+            bundles: Default::default(),
+            removed_components: Default::default(),
+            archetype_component_access: Default::default(),
+            main_thread_validator: Default::default(),
+            global_system_counter: Default::default(),
```

Comment:
I think we should consider starting `global_system_counter` with a value of 1. This would enable us to use the more natural default value of `0` for "last global_system_counter" values everywhere else.

Reply:
I really like this.

Reply:
Done in a8075a2118c34c2180db7cc47a2bb01941397a60.